state/executor.go

```diff
@@ -196,6 +203,16 @@ func (t *Transition) Write(txn *types.Transaction) error {
 		}
 	}
 
+	if gas := t.AvailableGas(); gas < TxGas {
+		t.logger.Info(
+			"Block doesn't have enough gas to process new transaction",
+			"remaining",
+			gas, "minimum required gas",
+			TxGas,
+		)
+		return ErrBlockLimitReached
+	}
+
```

Comment:
Why do we need this? Can't we rely on https://github.com/0xPolygon/polygon-sdk/blob/bfc7d94d077accabbbbb2d878574398453874e0a/state/executor.go#L379 ?

Reply:
You're right, the check you've highlighted is redundant because we have `subGasPool` when applying the txn, which literally does the same thing.

Fixed 👍 

48264b5